### ISSUE TITLE
Playing a key a second time does not work

### ISSUE THREAD

**Issue by trapicki** [lrq3000/starchord#1]
For me now, playing a key two times in a row does not give a sound at the second (and all following) strokes. If I play a different key, the "locked" key works again, but only once.
Hexiano worked fine for me, but after some update I cannot pin down this occured. Reinstall did not change the behaviour.

**Comment by 0x5c** [comment]
This bug is common to all the different versions of hexiano, from the original isokeys to branch of hexiano. 
Bug happens on all my devices. 
Tested on:
Oneplus One, cyanogen os 13.1 (Android 6.0.1)
Nexus7 2012 (grouper), CyanogenMod 12.1 (Android 5.1.1)

**Comment by lrq3000** [comment]
Oh that's really weird and unexpected, I specifically updated Hexiano to
avoid this kind of problem, because I need to be able to repeat notes
quickly for the songs I play.

Trapicki, can you tell me what version of Android you are using and what
device please?

2016-06-30 18:46 GMT+02:00 0x5c <notifications@github.com>:

**Comment by lrq3000** [comment]
Also, if you are an Android developper, you can plug the app to your
computer in debug mode, the debugger will display lots of information,
notably when you press a key, so you can know what happens. I guess there
will be an error raised by the sound manager.

2016-06-30 20:15 GMT+02:00 Stephen LARROQUE <lrq3000@gmail.com>:

**Comment by 0x5c** [comment]
Also, building the app upon the latest api level didn't fix it.

**Comment by lrq3000** [comment]
No, the app should be built with the API level I left, I tried to use a
more recent level but this would imply to revamp lots of the app to work
correctly (and also it would remove compatibility with older devices, as I
do possess ;) ).

But yeah if Trapicki also has Android > kitkat, then I guess something
changed with the sound manager.

0x5c since you can build the app, can you post the Logcat log here after
you press twice a button?

2016-06-30 21:17 GMT+02:00 0x5c <notifications@github.com>:

**Comment by trapicki** [comment]
I have an Moto G 2nd Gen with CyanogenMod 12.1 (Android 5.1) stable release. I could provide debug data if you tell me what you need and pointers how to get it.

**Comment by phnord** [comment]
I'm on stock Nougat on Google Nexus 6, and can confirm that most notes will not retrigger after the first time they're played, which makes this unfortunately pretty unusable :/

**Comment by vimino** [comment]
Hi there. On a _ZTE Ion Gold_ and having installed Hexiano through _FDroid_, I noticed a similar issue as keys stopped playing sounds when pressed more than once.
[I did a quick test using logcat](https://github.com/lrq3000/hexiano/files/1023352/hexiano_logcat.txt) where I pressed different keys, followed by the same one repeatedly. Effectively reproducing the problem as no sound was played after the first press.
It seems that the message "**E/AudioPolicyManager(  308): unknown stream type 13**" kept repeating close to the end, when it happened.

**Comment by lrq3000** [comment]
Thanks @vimino for the logcat, indeed this seems to indicate that Hexiano cannot change the stream parameters (volume, pitch, etc.), so the sound cannot be played.

I notice also the following lines:

```
W/NuPlayerRenderer(  308): AudioSink write short frame count 0 < 1408
E/AudioCache(  308): Heap size overflow! req size: 1049984, max size: 1048576
```

Which happen before any user event in Hexiano. This might indicate a lack of memory when loading the samples (which is the first thing done when launching Hexiano).

The first thing I would advise would be to increase the cache settings in Hexiano options (only possible in my version - please install from this github [using this apk](https://github.com/lrq3000/hexiano/releases/tag/v1.0.2)).

If this does not work, then it might be [a problem of permissions which I would need to update](https://stackoverflow.com/questions/3536391/how-to-correctly-set-mediaplayer-audio-stream-type).

If this fails again, then I would need more logcat info at this stage, using [for example this app](https://github.com/nolanlawson/Catlog).

Let me know if someone with this issue try to raise the cache if it does work or not!

**Comment by vimino** [comment]
The extended version (Hexiano Extended v1.0.2) is playing the same key properly. Tested without changing any setting.

**Comment by 0x5c** [comment]
@lrq3000 The apk does not install

**Comment by lrq3000** [comment]
@vimino Great news! Thanks for the feedback!
@0x5c What error are you getting?

**Comment by 0x5c** [comment]
@lrq3000 I was able to install it after deleting the version I already had. I checked the version numbers, and they where the same.

I can't say if the bug is fixed, because it always took about 5-15 minutes for it to show up. I'll try to get it some real testing going and we'll see what happens.

**Comment by jaxter184** [comment]
also works for me on android 7.1.1

**Comment by lrq3000** [comment]
@0x5c Note that the APK found in this Github repository cannot be found anywhere else (not on Fdroid nor Google Play store) since I am not the owner of these. Thus if you installed from another source and the version number is the same, it might well be because I did a mistake in the code and I should update :-)

It was planned to change the name of this repo to avoid confusion with the original Hexiano, but I am now facing major limitations with Java so I am thinking about rewriting from scratch using another framework such as Kivy (but don't expect any release anytime soon!).

**Comment by 0x5c** [comment]
@lrq3000 I am pretty sure that the version I had came from your repo.
Sorry for not testing the app yet, my time was stolen by christmas.

**Comment by trapicki** [comment]
Just when things got rolling again a few days ago, I bootlooped my phone, had to reflash the system, and decided to upgrade from Cyanogen 12 to LineageOS 14.1. Sorry, cannot rebuild the original setup for comparison.

One important thing that I have seen when reinstalling programs is, that I was maybe talking about the version 0.82.1 ALPHA that I found on F-Droid _Archive_, which is 5 years old.

1.0.2 works perfect for me now. 

The original being now not maintained any more, and your repo being the actively supported, I suggest you add your version to F-Droid again. Hexiano is great, it deserves more publicity!

**Comment by lrq3000** [comment]
Thanks for all the feedback guys, that's great news!

@trapicki About uploading to F-Droid, as I am not the original author of Hexiano, I am not sure it is possible. We discussed a few years ago with the author, but he did not want to continue the project under an opensource license :-/ So for the moment the project is stuck in this state until I restart from scratch...

**Comment by trapicki** [comment]
@lrq3000: Regarding publishing:
Hexiano is a continuation of IsoKeys, both licensed under GLP3+. This means any anyone is free to run, study, change, and finally redistribute it (under the same conditions.)

IsoKeys is a registered trade mark, like Hexiano and the Hexiano logo. My current understanding is that both programs are distributed via Google Play Store for free, with possible in-app payments for extra features. Both versions seem to be not in active development since 2012.

If the trademark owners do not give permission to use their intellectual property - which is their good right - let's just rename it again, give all fair and required credit, and publish it. In the end that is why the software was published under GPL in the first place.
I can think of something like "Hexagonal Piano", "Isomorphic Keyboard" or any combination thereof.

**Comment by vimino** [comment]
Yes, that would seem to be the case. If all else fails, fork it!
It could even be named "Isogonal Harmonics" and it's a good opportunity to explore/update frameworks, if the owner has time for that.

**Comment by lrq3000** [comment]
@trapicki Yes you're totally correct, only the logo and name are trademarked, if I change I could publish by myself. But i'm not too good in logo design and name branding, so if you guys have any idea of names and logos, feel free to contribute, that would be awesome :-D

Just a note about the name, I would prefer something short, like 2 or 3 syllables maximum, so people can easily remember the name. Something like isoboard, keygon, etc.

**Comment by trapicki** [comment]
Proposal for a logo in issue #2.

**Comment by brunetton** [comment]
Could you please provide an APK file in releases ? Unfortunately the last published version ([1.0.2](https://github.com/lrq3000/hexiano/releases/tag/v1.0.2 )) doesn't work anymore on my new phone (Android 14).
Of course, #3 would be the better world but a downloadeable APK would really be appreciated (as I'm not an Androids dev, it's quite long to install all the dev dependencies and make it work)
Thank you a lot!

**Comment by lrq3000** [comment]
Yes all of these is still in the works, should be done soon.

You can download a debug apk here : 
https://github.com/lrq3000/starchord/actions/runs/21780956373/artifacts/5417159141

**Comment by brunetton** [comment]
My bad, I didn't looked at the pipelines 😔

Wow, you can't imagine how much I feel good to finally being able to play Janko piano again on my phone ❤️
Thank you 🙏

**Comment by lrq3000** [comment]
No no don't apologize, I planned to make a release quickly after closing this issue but life got in the way so I am postponing for a few days but it will get done in the upcoming days/weeks :-)

Thank you very much for your patience and your very helpful feedback and for still being interested in this project after so many years, it is very heartwarming 😊
